### ISSUE TITLE
Rails 6.1 support

### ISSUE THREAD

**Issue by kirillplatonov** [Shopify/shopify_app#1163]
Since v17 `shopify_app` forces to use Rails < 6.1.
![CleanShot 2021-01-26 at 20 23 41@2x](https://user-images.githubusercontent.com/839922/105880516-7d232600-6014-11eb-9365-d1c43923b827.jpg)

It's stated in Readme that it's not compatible due to changes in `cookies_same_site_protection` setting in Rails 6.1. It's very easy to fix by restoring an old value of this setting (as a temporary workaround):
```ruby
# config/application.rb

config.load_defaults 6.1
config.action_dispatch.cookies_same_site_protection = nil
```

Can we allow installing `shopify_app` in Rails 6.1 and in Readme mention this workaround instead of forcing to use Rails 6.0? For now, the only solution is forking and regularly pulling the latest commits from upstream.

**Comment by tanema** [comment]
There are more issues specifically around asset pipeline with 6.1 as well beyond just samesite. Our devs are looking into the best way to handle this.

**Comment by oliwoodsuk** [comment]
I'm using Rails 6.1 and it's working fine (seemingly).
I really want to upgrade to the latest shopify_app version, as some customers are having issues with Safari and JWT. Think I'll hold off though until Shopify say 6.1 is all good.

**Comment by awd** [comment]
What's the latest info on this? Still not supporting the latest Rails 6.1?

**Comment by netwire88** [comment]
Hi @oliwoodsuk, @awd , we are having the same issues. Rails 6.1 supports disregarding arguments for webhooks, but we can't use it since shopify_app doesn't support Rails 6.1

**Comment by oliwoodsuk** [comment]
> Hi @oliwoodsuk, @awd , we are having the same issues. Rails 6.1 supports disregarding arguments for webhooks, but we can't use it since shopify_app doesn't support Rails 6.1

Hi @netwire88 , we're using version 16.1 of the shopify_app gem with Rails 6.1 and it's working fine (well for all the things we use it for). I think this incompatibility issue is related to the more recent releases.

`gem 'shopify_app', '~> 16.1.0'`

**Comment by itsniper** [comment]
Not ideal, but for now I've forked the project and [reverted](https://github.com/JDSoftwareCO/shopify_app/commit/54063698316bd6c310c41e212550f9b1d4617805) the Rails 6.1 limitation. As @kirillplatonov mentioned, I also added `config.action_dispatch.cookies_same_site_protection = nil` to `config/application.rb`.

My fork is currently at v17.0.0 with no issues that I've noticed. I'm also using JWT and not session cookies for auth, which I think makes a difference.
